theano/tensor/signal/downsample.py

```diff
@@ -343,17 +366,39 @@ def make_node(self, x, maxout, gz):
     def perform(self, node, inp, out):
         x, maxout, gz = inp
         gx_stg, = out
-        gx = numpy.zeros_like(x)
-
+        
         #number of pooling output rows
         pr = maxout.shape[-2]
         #number of pooling output cols
         pc = maxout.shape[-1]
         ds0, ds1 = self.ds
         st0, st1 = self.st
-        img_rows = x.shape[-2]
-        img_cols = x.shape[-1]
-
+        img_rows = x.shape[-2] + 2 * self.padding[0]
+        img_cols = x.shape[-1] + 2 * self.padding[1]
+        pad_h = self.padding[0]
+        pad_w = self.padding[1]
```

Comment:
You are still using `self.padding[0]` and `1`, and then defining shorter variables for them.
You should define `pad_h`, and *then* use it in `img_rows`.